etc/kayobe/inventory/group_vars/all/firewall

```diff
@@ -0,0 +1,437 @@
+  {% if network | net_zone %}
+  {% set _ = network_zones.append({'zone': network | net_zone }) %}
+  {% endif %}
+  {% endfor %}
+  {{ network_zones }}
+
+# Rules are filtered to ensure that none are conflicting
+stackhpc_firewalld_rules: |
+  {% set stackhpc_firewalld_rules_filtered = [] %}
+  {% for rule in stackhpc_firewalld_rules_unfiltered %}
+  {% if rule | ansible.utils.remove_keys('state') not in stackhpc_firewalld_rules_filtered | map('ansible.utils.remove_keys', 'state') %}
+  {% set _ = stackhpc_firewalld_rules_filtered.append(rule) %}
+  {% endif %}
+  {% endfor %}
+  {{ stackhpc_firewalld_rules_filtered }}
+
+stackhpc_firewalld_rules_unfiltered: |
+  {{  (stackhpc_controller_firewalld_rules if 'controllers' in group_names else []) +
+      (stackhpc_compute_firewalld_rules if 'compute' in group_names else []) +
+      (stackhpc_storage_firewalld_rules if 'storage' in group_names else []) +
+      (stackhpc_monitoring_firewalld_rules if 'monitoring' in group_names else []) +
+      (stackhpc_seed_firewalld_rules if 'seed' in group_names else []) +
+      (stackhpc_seed_hypervisor_firewalld_rules if 'seed-hypervisor' in group_names else []) +
+      (stackhpc_wazuh_manager_infra_vm_firewalld_rules if 'wazuh-manager' in group_names else []) +
+      (stackhpc_ansible_control_infra_vm_firewalld_rules if 'ansible-control' in group_names else []) }}
+
+###############################################################################
+# Controller firewalld rules
+
+stackhpc_controller_firewalld_rules: >-
+  {{ stackhpc_controller_firewalld_rules_default |
+    selectattr('enabled', 'true') |
+    map(attribute='rules') |
+    flatten |
+    selectattr('network', 'in', network_interfaces) |
+    selectattr('zone') |
+    union(stackhpc_controller_firewalld_rules_extra) |
+    unique |
+    select }}
+
+stackhpc_controller_firewalld_rules_default:
+  # Common
+  - rules:
+      - service: ssh
+        zone: "{{ admin_oc_net_name | net_zone }}"
+        network: "{{ admin_oc_net_name }}"
+        state: enabled
+      - service: dhcp
+        zone: "{{ provision_wl_net_name | net_zone }}"
+        network: "{{ provision_wl_net_name }}"
+        state: enabled
+      - service: ntp
+        zone: "{{ provision_wl_net_name | net_zone }}"
+        network: "{{ provision_wl_net_name }}"
+        state: enabled
+      - service: tftp
+        zone: "{{ provision_wl_net_name | net_zone }}"
+        network: "{{ provision_wl_net_name }}"
+        state: enabled
+      - service: cockpit
+        zone: "{{ public_net_name | net_zone }}"
+        network: "{{ public_net_name }}"
+        state: disabled
+      - service: dhcpv6-client
+        zone: "{{ public_net_name | net_zone }}"
+        network: "{{ public_net_name }}"
+        state: disabled
+    enabled: true
+  - rules:
+      - service: ssh
+        zone: "{{ public_net_name | net_zone }}"
+        network: "{{ public_net_name }}"
+        state: disabled
```

Comment:
I suppose that strictly these rules are for when you use the default `public` zone, and could apply to any network. Maybe the correct condition is that `public` is one of the zones referenced in the config on this host, and for the SSH rule it should also not be the admin_oc_net_name

Reply:
I'm wondering if it would help to have a list of common rules?

Reply:
Definitely agree with the common rules idea.
I'm not sure I 100% understand what you mean with your other comment, but I've updated the config with my interpretation